pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -94,29 +93,47 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 				}
 			}
 
+			// if a Pod is terminating already, we do not care it being not ready because this is expected to be the case
+			if pod.DeletionTimestamp != nil {
```

Comment:
We want to report the error though during the exit no?

Reply:
I am actually not anymore sure if this is needed. What I had in mind was the following:

* a Pod is created and runs fine
* at some point the user-container goes out of memory and is restarted, from now on it has `LastTerminationState.Terminated` set.
* And now it runs fine, an hour, a day maybe.
* And then it gets scaled down.
* The loop runs.

The Pod will not be ready so it will check the container status. Idea was to prevent reporting a failure that happened long time ago.

Though, the only relevant case would be something like a scaleDown from 2 to 1 of the ready Pod that had this OOM long time ago which the other Pod must be down so that the deployment has zero available replicas. Is very unlikely to happen and definitely not a homogenous state of the workload across pods.